hw2/04-background.js

```diff
@@ -1,13 +1,24 @@
-const interval = document.getElementById('number');
+const interval = document.getElementById('number').value;
 const submit = document.getElementById('submit');
-const background = document.getElementById('background');
+// const background = document.getElementById('background');
 // Add your code here
 
-submit.addEventListener('click', (event) => {
-    setInterval(changeColor, interval*1000);
-})
+// https://stackoverflow.com/questions/1484506/random-color-generator
+const randomColor = function generateRandomRGB() {
+  const letters = '0123456789ABCDEF';
+  let color = '#';
+  for (let i = 0; i < 6; i += 1) {
+    color += letters[Math.floor(Math.random() * 16)];
+  }
+  color += '80';
+  return color;
+};
 
-changeColor = function changeBackgroundColorEveryInterval() {
-    background.innerHTML
-    
-}
\ No newline at end of file
+const changeColor = function changeBackgroundColorEveryInterval() {
+  document.body.style.background = randomColor();
+};
+
+submit.addEventListener('click', () => {
+  console.log(interval);
+  setInterval(changeColor, Number(interval * 1000));
```

Comment:
By default your program should change background color every 3 seconds when the screen is loaded. You might want to consider using the setInterval function setting up default behaviour.

Reply:
gotcha, might be good for functional abstraction as well